src/controller/CHIPDeviceController.cpp

```diff
@@ -1765,51 +1534,41 @@ void DeviceCommissioner::PerformCommissioningStep(DeviceProxy * proxy, Commissio
         // TODO(cecille): Set time as well once the time cluster is implemented
         // TODO(cecille): Worthwhile to keep this around as part of the class?
         // TODO(cecille): Where is the country config actually set?
-        ChipLogProgress(Controller, "Setting Regulatory Config");
-        uint8_t regulatoryLocation = EMBER_ZCL_REGULATORY_LOCATION_TYPE_OUTDOOR;
-#if CONFIG_DEVICE_LAYER
-        CHIP_ERROR status = DeviceLayer::ConfigurationMgr().GetRegulatoryLocation(regulatoryLocation);
-#else
-        CHIP_ERROR status = CHIP_ERROR_NOT_IMPLEMENTED;
-#endif
-        if (status != CHIP_NO_ERROR)
-        {
-            ChipLogError(Controller, "Unable to find regulatory location, defaulting to outdoor");
-        }
-
+        uint8_t regulatoryLocation                  = EMBER_ZCL_REGULATORY_LOCATION_TYPE_OUTDOOR;
         static constexpr size_t kMaxCountryCodeSize = 3;
         char countryCodeStr[kMaxCountryCodeSize]    = "WW";
         size_t actualCountryCodeSize                = 2;
 
+        bool hasRegulatoryLocation = false;
+        bool hasCountryCode        = false;
 #if CONFIG_DEVICE_LAYER
-        status = DeviceLayer::ConfigurationMgr().GetCountryCode(countryCodeStr, kMaxCountryCodeSize, actualCountryCodeSize);
-#else
-        status = CHIP_ERROR_NOT_IMPLEMENTED;
+        hasRegulatoryLocation = (CHIP_NO_ERROR == DeviceLayer::ConfigurationMgr().GetRegulatoryLocation(regulatoryLocation));
+        hasCountryCode =
+            (CHIP_NO_ERROR ==
+             DeviceLayer::ConfigurationMgr().GetCountryCode(countryCodeStr, kMaxCountryCodeSize, actualCountryCodeSize));
 #endif
-        if (status != CHIP_NO_ERROR)
+
+        if (hasRegulatoryLocation)
+        {
+            ChipLogError(Controller, "Unable to find regulatory location, defaulting to outdoor");
+        }
+
+        if (hasCountryCode)
         {
             ChipLogError(Controller, "Unable to find country code, defaulting to WW");
         }
-        chip::CharSpan countryCode(countryCodeStr, actualCountryCodeSize);
 
-        GeneralCommissioningCluster genCom;
-        genCom.Associate(proxy, 0);
-        genCom.SetRegulatoryConfig(mSuccess.Cancel(), mFailure.Cancel(), regulatoryLocation, countryCode, breadcrumb,
-                                   kCommandTimeoutMs);
+        GeneralCommissioning::Commands::SetRegulatoryConfig::Type request;
+        request.location    = static_cast<EmberAfRegulatoryLocationType>(regulatoryLocation);
```

Comment:
This needs to cast to the right type (the cluster-objects one) to compile on tip....  `EmberAfRegulatoryLocationType` is gone.

Reply:
I have reverted all those changes.